### ISSUE TITLE
usd-core failing on Windows + conda env + py3.9

### ISSUE THREAD

**Issue by Caenorst** [PixarAnimationStudios/USD#1602]
### Description of Issue
Hi, I tested on two different machines to install and import `usd-core` pypi package and it's failing with the following logs:

```
test_usd.py:21: in <module>
    from pxr import Usd, UsdGeom
C:\Users\myusername\Anaconda3\envs\kaolin_release\lib\site-packages\usd_core-21.8-py3.9-win-amd64.egg\pxr\Usd\__init__.py:24: in <module>
    from pxr import Tf
C:\Users\myusername\Anaconda3\envs\kaolin_release\lib\site-packages\usd_core-21.8-py3.9-win-amd64.egg\pxr\Tf\__init__.py:163: in <module>
    PreparePythonModule()
C:\Users\myusername\Anaconda3\envs\kaolin_release\lib\site-packages\usd_core-21.8-py3.9-win-amd64.egg\pxr\Tf\__init__.py:88: in PreparePythonModule
    module = importlib.import_module(
C:\Users\myusername\Anaconda3\envs\kaolin_release\lib\importlib\__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ImportError: DLL load failed while importing _tf: The specified module could not be found.
```

#### Remarks
On the same machine there is no problem with `virtualenv`, neither with python version 3.7.

### Steps to Reproduce
1. Install conda
2. Make a conda environment with python 3.9 (I believe it's also failing on 3.8)

```
conda create -n myenv python=3.9
conda activate myenv
```

3. pip install usd-core
4. python -c "from pxr import Usd, UsdGeom"

### System Information (OS, Hardware)
Windows 10

### Package Versions
usd-core == 21.8
conda, python == 3.9

**Comment by rstelzleni** [comment]
Starting with python 3.8 on Windows the logic for finding dlls in the path changed. We had written code to handle this for the python packages, but it sounds like conda is exposing some flaw in that solution. I'm able to reproduce this in Windows conda, with the python 3.8 and 3.9 packages.

**Comment by rstelzleni** [comment]
I haven't been able to completely get to the bottom of this yet, but I can report that setting the environment variable `CONDA_DLL_SEARCH_MODIFICATION_ENABLE` to 1 makes the package work for me. Just randomly googling that env var, it seems like some packages work when this is set but others may not. So, your mileage may vary, but this could be a way to get this working while we look for a permanent fix.

**Comment by jilliene** [comment]
Filed as internal issue #USD-6866

**Comment by sunyab** [comment]
Closing out old issue, this should be addressed by #1642 which was merged in the 22.03 release.
